torch/_dynamo/config_utils.py

```diff
@@ -19,36 +24,84 @@ def install_config_module(module):
     class ConfigModuleInstance(ConfigModule):
         _bypass_keys = set()
 
-    def visit(source, dest, prefix):
-        """Walk the module structure and move everything to module._config"""
+    def visit(source, dest, prefix, ignore_all=False):
+        """
+        Walk the module structure and move everything to
+        module._config and module._compile_ignored
+        """
+        ignored = get_assignments_with_compile_ignored_comments(source)
         for key, value in list(source.__dict__.items()):
             if key.startswith("__") or isinstance(value, (ModuleType, FunctionType)):
                 continue
 
             name = f"{prefix}{key}"
             if isinstance(value, CONFIG_TYPES):
-                config[name] = value
+                if ignore_all or key in ignored:
+                    compile_ignored[name] = value
+                else:
+                    config[name] = value
                 default[name] = value
                 if dest is module:
                     delattr(module, key)
             elif isinstance(value, type):
                 assert value.__module__ == module.__name__
                 # a subconfig with `class Blah:` syntax
                 proxy = SubConfigProxy(module, f"{name}.")
-                visit(value, proxy, f"{name}.")
+                visit(value, proxy, f"{name}.", ignore_all=name in ignored)
                 setattr(dest, key, proxy)
             else:
                 raise AssertionError(f"Unhandled config {key}={value} ({type(value)})")
 
     config = dict()
+    compile_ignored = dict()
     default = dict()
+
     visit(module, module, "")
     module._config = config
+    module._compile_ignored = compile_ignored
     module._default = default
-    module._allowed_keys = set(config.keys())
+
+    config_keys, compile_ignored_keys = set(config.keys()), set(compile_ignored.keys())
+    assert config_keys.isdisjoint(compile_ignored_keys)
+    module._allowed_keys = config_keys | compile_ignored_keys
+    module._compile_ignored_keys = set(compile_ignored.keys())
+
     module.__class__ = ConfigModuleInstance
 
 
+# Gets all the keys (i.e. assignments) with a @compile_ignored comment
+def get_assignments_with_compile_ignored_comments(module):
+    source_code = inspect.getsource(module)
+    assignments = set()
+
+    # Tokenize the source code to retrieve comments
+    tokens = tokenize.tokenize(io.BytesIO(source_code.encode("utf-8")).readline)
+    current_comment = "", -1
+    prev_name = ""
+    prev_assigned = "", -1
+
+    for token in tokens:
```

Comment:
So, looking at this code, it seems like it's easy to accidentally tag the "wrong" name as ignored, because there is potentially a fairly long distance in which a name assignment can get associated with a current comment.

There are a few things we could do, but I think the simplest thing is to just make the comments actually contain the name of the config option they apply to, and then check that the inferred name matches what the user wrote in.

If you dislike the duplication, I would prefer it if we explicitly "reset" the current comment once you hit important tokens (like newline marker), so there is a clear delineation of the scope.

Reply:
> because there is potentially a fairly long distance in which a name assignment can get associated with a current comment.

In reality, it's only:
1. previous line comment
2. same line comment

I can add a bunch of tests to prove that it doesn't get murky

Reply:
It would be good to raise an error if a `@compile_ignored` comment isn't attached to any token

Reply:
done